community/kernel/src/main/java/org/neo4j/kernel/recovery/Recovery.java

```diff
@@ -38,116 +37,94 @@
  */
 public class Recovery extends LifecycleAdapter
 {
-    public interface Monitor
```

Comment:
I'm not a fan of moving out interfaces that are very tied to its parent outside. Why this refactoring?